core/dnsserver/server.go

```diff
@@ -155,10 +151,7 @@ func (s *Server) Address() string { return s.Addr }
 // defined in the request so that the correct zone
 // (configuration and middleware stack) will handle the request.
 func (s *Server) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
-	s.serveDNSWithContext(context.Background(), w, r)
-}
-
-func (s *Server) serveDNSWithContext(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) {
+	ctx := context.Background()
```

Comment:
I think we *can* do this simpler. The coredns.Server needs a dns.ReadDecorator to do some gRPC stuff while reading the packet. The writing can be done like it is done now in the coredns/grpc.
Keeping the Context can be done with using a function litteral when setting up the handler, where we pull out the context from gRPC and call ServeDNS.
We should also fix ServeDNS to take a context.Context. I'll make some changes that show what I mean here.